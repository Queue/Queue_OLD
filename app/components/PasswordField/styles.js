//
// PasswordField styles.js

import { StyleSheet } from 'react-native';
import Colors from '../../lib/colors';
import Fonts from '../../lib/fonts';

const inputHeight = 40;

export default styles = {
  passwordField(textColor) {
    return {
      fontFamily: Fonts.content,
      color: textColor,
      margin: 10,
      paddingLeft: 15,
      paddingRight: 15,
      height: inputHeight,
      justifyContent: 'center',
      alignItems: 'center',
      borderRadius: 2,
      backgroundColor: 'white',
      zIndex: 1
    }
  },
  showText() {
    return {color: Colors.primaryForeground}
  },
  showTextHighlight() {
    return {
      position: 'absolute',
      right: 20,
      top: (inputHeight / 2),
      zIndex: 9999,
      backgroundColor: 'transparent'
    }
  }
};
