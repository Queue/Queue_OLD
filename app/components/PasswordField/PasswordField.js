//
// PasswordField Component

import React, { Component } from 'react';
import { TextInput, View, TouchableHighlight, Text } from 'react-native';
import styles from './styles';
import Colors from '../../lib/colors';

export default class PasswordField extends Component {
  constructor(props) {
    super(props);

    this.state = {
      hide: true
    };
  }

  showPass() {
    this.setState({
      hide: !this.state.hide
    });
  }

  render() {
    let eyeOfRa = '\uD83D\uDC41',
        change = this.props.change,
        placeholder = this.props.placeholder,
        edit = this.props.edit,
        val = this.props.val,
        textColor = this.props.textColor,
        hide = this.state.hide;

    return (
      <View>
        <TextInput
          style = {styles.passwordField(textColor)}
          onChangeText = {change}
          placeholder = {placeholder}
          editable = {edit}
          value = {val}
          autoCapitalize = {'none'}
          secureTextEntry = {hide}
        />
        <TouchableHighlight
          style = {styles.showTextHighlight()}
          onPress = {this.showPass.bind(this)}
          underlayColor = {'transparent'}>
          <Text style={styles.showText()}>{eyeOfRa}</Text>
        </TouchableHighlight>
      </View>
    );
  }
}

PasswordField.propType = {
  placeholder: React.PropTypes.string,
  change: React.PropTypes.func,
  val: React.PropTypes.string,
  textColor: React.PropTypes.string,
  secure: React.PropTypes.bool
}

PasswordField.defaultProps = {
  placeholder: 'Placeholder',
  textColor: Colors.primaryForeground,
  editable: false,
  secure: true
};
