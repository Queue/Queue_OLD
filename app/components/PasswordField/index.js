//
// PasswordField index.js

import PasswordField from './PasswordField';
export default PasswordField;
