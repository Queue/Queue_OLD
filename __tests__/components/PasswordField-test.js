//
// PasswordField component test

import React from 'react';
import PasswordField from '../../app/components/PasswordField';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <PasswordField
      placeholder='test'
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
