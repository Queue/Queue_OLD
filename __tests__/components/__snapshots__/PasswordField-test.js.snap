exports[`test renders correctly 1`] = `
<View>
  <TextInput
    autoCapitalize="none"
    editable={undefined}
    onChangeText={undefined}
    placeholder="test"
    secureTextEntry={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 2,
        "color": "#48C948",
        "fontFamily": "Helvetica Neue",
        "height": 40,
        "justifyContent": "center",
        "margin": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "zIndex": 1,
      }
    }
    value={undefined} />
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "backgroundColor": "transparent",
          "position": "absolute",
          "right": 20,
          "top": 20,
          "zIndex": 9999,
        },
      ]
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#48C948",
        }
      }>
      👁
    </Text>
  </View>
</View>
`;
